move details on channels into the dedicated page (#11434)
NixOS/nix: doc/manual/src/command-ref/files/channels.md

## Channels

Channels are a mechanism for obtaining Nix expressions over the web.

The moving parts of channels are:
- The official channels listed at <https://nixos.org/channels>
- The user-specific list of [subscribed channels](#subscribed-channels)
- The [downloaded channel contents](#channel-contents)
- The [Nix expression search path](@docroot@/command-ref/conf-file.md#conf-nix-path)

> **Note**
>

> > The state of a subscribed channel is external to the Nix expressions relying on it.
> > This may limit reproducibility.

Reviewer: We can confidently state that this is a problem for reproducibility and make it slightly more actionable.

```suggestion
> The state of a subscribed channel is external to the Nix expressions relying on it.
> This limits reproducibility, and may be a reason to use a lock file instead of channels.
```

I would like to link a resource that helps with deciding the kind of lock file, but I don't want to derail this review, so I'd prefer to settle for an incremental improvement here, as far as my input is concerned. I'd be happy to accept a suggestion.

Author: We'd be opening up quite the rabbit hole here. I generally agree with the direction, but my aspiration would be to point people to The Right Way of doing things, and that to my knowledge still doesn't exist. The current version says something that is at least not wrong: use fetchers explicitly.

Reviewer: I guess a pin is not necessarily in a lock file. Fair enough.
We could confidently say that pinning and/or locking is The Right Way, but it might be a distraction.

Initially I thought nix.dev had a suitable article about this, but they seem to be about specific solutions, which is good for tutorials; was looking for a guide.

Let's not get ourselves stuck on this and fill this gap later.

>

> > Dependencies on other Nix expressions can be declared explicitly with:
> > - [`fetchurl`](@docroot@/language/builtins.md#builtins-fetchurl), [`fetchTarball`](@docroot@/language/builtins.md#builtins-fetchTarball), or [`fetchGit`](@docroot@/language/builtins.md#builtins-fetchGit) in Nix expressions

Reviewer: ```suggestion
```

This is going off topic. The answer to "how do I fetch?" should probably not be on the Channels page, and mention all options, not just the eval-time fetchers.
- nix-channel
- FODs, best fed to derivations
- builtin fetchers, best fed to `import` and friends

Author: There's https://github.com/NixOS/nix.dev/issues/750 to address this, which needs some focus time to write down cleanly. I agree this would not be about channels but essentially a concept page.

> > - the [`-I` option](@docroot@/command-ref/opt-common.md#opt-I) in command line invocations for explicitly determining the value of [lookup paths](@docroot@/language/constructs/lookup-path.md)

Reviewer: ```suggestion
> - the [`-I` option](@docroot@/command-ref/opt-common.md#opt-I) in command line invocations for explicitly setting the value of [lookup paths](@docroot@/language/constructs/lookup-path.md)
```

"Determine" can mean to observe and draw a conclusion.

If we need to be more vague

```suggestion
> - the [`-I` option](@docroot@/command-ref/opt-common.md#opt-I) in command line invocations for explicitly modifying the value of [lookup paths](@docroot@/language/constructs/lookup-path.md)
```

However, I now doubt that this should even be here, because command line invocations are transient - not _declaring explicitly_. A script might qualify, but the current text is about the command line (ie interactive).

Also you'd have to be explicit about what kind of `-I` value is actually qualifies as a precise pin for the purpose of eval reproducibility.

```suggestion
```

Author: Let's just leave the pointer here for now and address all of the details in a dedicated page.

## Subscribed channels

The list of subscribed channels is stored in a file:

- `~/.nix-channels`
- `$XDG_STATE_HOME/nix/channels` if [`use-xdg-base-directories`](@docroot@/command-ref/conf-file.md#conf-use-xdg-base-directories) is set to `true`

Each line maps a channel name to a URL in the following format:

```
<url> <name>
```

## Channels contents

The [`nix-channel`](@docroot@/command-ref/nix-channel.md) command uses a [profile](@docroot@/command-ref/files/profiles.md) to keep track of channels:

> - `$XDG_STATE_HOME/nix/profiles/channels` for regular users
> - `$NIX_STATE_DIR/profiles/per-user/root/channels` for `root`

Reviewer: ```suggestion
- profile `$XDG_STATE_HOME/nix/profiles/channels` for regular users
- profile `$NIX_STATE_DIR/profiles/per-user/root/channels` for `root`
```

Author: I'm not sure what this is supposed to achieve
